virtcontainers/cgroups_test.go

```diff
@@ -224,3 +224,62 @@ func TestIsSystemdCgroup(t *testing.T) {
 		assert.Equal(t.expected, isSystemdCgroup(t.path), "invalid systemd cgroup path: %v", t.path)
 	}
 }
+
+func TestValidCgroupPath(t *testing.T) {
+	assert := assert.New(t)
+	var err error
+	var path string
+
+	renamedCgroupPath, err := renameCgroupPath(defaultCgroupPath)
+	assert.NoError(err)
+
+	for _, t := range []struct {
+		path     string
+		expected string
+	}{
+		{"../../../", renamedCgroupPath},
+		{"../", renamedCgroupPath},
+		{".", renamedCgroupPath},
+		{"../../../", renamedCgroupPath},
+		{"./../", renamedCgroupPath},
+	} {
+		path, err = validCgroupPath(t.path, false)
+		assert.NoError(err)
+		assert.Equal(t.expected, path)
+	}
+
+	renamedCgroupPath, err = renameCgroupPath("foo")
+	assert.NoError(err)
+	path, err = validCgroupPath("../../../foo", false)
+	assert.NoError(err)
+	assert.Equal(filepath.Join(defaultCgroupPath, renamedCgroupPath), path)
+
+	renamedCgroupPath, err = renameCgroupPath("/hi")
+	assert.NoError(err)
+	path, err = validCgroupPath("/../hi", false)
+	assert.NoError(err)
+	assert.Equal(renamedCgroupPath, path)
+
+	renamedCgroupPath, err = renameCgroupPath("/hi/foo")
+	assert.NoError(err)
+	path, err = validCgroupPath("/../hi/foo", false)
+	assert.NoError(err)
+	assert.Equal(renamedCgroupPath, path)
+
+	renamedCgroupPath, err = renameCgroupPath("o / g")
+	assert.NoError(err)
+	path, err = validCgroupPath("o / m /../ g", false)
+	assert.NoError(err)
+	assert.Equal(filepath.Join(defaultCgroupPath, renamedCgroupPath), path)
+
+	_, err = validCgroupPath("o / m /../ g", true)
```

Comment:
nit: these tests are a lot more readable if you have a brief comment/description for each test case. ie, this could be "invalid systemd cgroup path". The first few are really just testing filepath.Clean, right?

Thanks for adding all the tests.